### PR TITLE
create NAT gateways for private subnets

### PR DESCRIPTION
**Pull request by hakamadare**
create a NAT gateway in each public subnet, create a route table with
the respective NAT gateway as its default route, associate the
appropriate route table with each private subnet

also adding a `.gitignore` to facilitate testing

### CODE REVIEW

**Comment by hakamadare** [comment]
@antonbabenko i've pushed some additional changes; there's now an `enable_nat_gateway` variable, which is "false" by default, and it toggles the creation and configuration of NAT Gateways on and off.

File: main.tf [hunk 1]
```diff
@@ -31,12 +31,22 @@ resource "aws_route" "public_internet_gateway" {
   gateway_id             = "${aws_internet_gateway.mod.id}"
 }
 
+resource "aws_route" "private_nat_gateway" {
+  route_table_id         = "${element(aws_route_table.private.*.id, count.index)}"
+  destination_cidr_block = "0.0.0.0/0"
+  nat_gateway_id         = "${element(aws_nat_gateway.natgw.*.id, count.index)}"
+  count                  = "${length(var.private_subnets) * lookup(map(var.enable_nat_gateway, 1), "true", 0)}"
+
+  depends_on             = ["aws_nat_gateway.natgw"]
```
Comment (antonbabenko) [inline main.tf]: I think this `depends_on` is not required and can be removed.

File: outputs.tf [hunk 2]
```diff
@@ -11,11 +11,11 @@ output "vpc_id" {
...
 
 output "public_route_table_id" {
-  value = "${aws_route_table.public.id}"
+  value = "${join(",", aws_route_table.public.*.id)}"
 }
 
 output "private_route_table_id" {
-  value = "${aws_route_table.private.id}"
+  value = "${join(",", aws_route_table.private.*.id)}"
 }
```
Comment (antonbabenko) [inline outputs.tf]: could you also add outputs of `aws_eip.nateip` as it may be useful?

File: outputs.tf [hunk 3]
```diff
@@ -11,11 +11,11 @@ output "vpc_id" {
...
 
 output "public_route_table_id" {
-  value = "${aws_route_table.public.id}"
+  value = "${join(",", aws_route_table.public.*.id)}"
 }
 
 output "private_route_table_id" {
-  value = "${aws_route_table.private.id}"
+  value = "${join(",", aws_route_table.private.*.id)}"
```
Comment (antonbabenko) [inline outputs.tf]: I think we should return native lists (as on line 2,6), instead of using join/split.

File: main.tf (hunk 1 above)
Comment (hakamadare) [inline main.tf]: done.

File: outputs.tf (hunk 3 above)
Comment (hakamadare) [inline outputs.tf]: ok.  i was afraid that changing the type returned by the output (from string to list) would be a backwards-incompatible change; i propose to change the output names from _e.g._ `private_route_table_id` (string) to `private_route_table_ids` (list), is that ok?

File: outputs.tf (hunk 2 above)
Comment (hakamadare) [inline outputs.tf]: done, and updated README.

File: outputs.tf (hunk 3 above)
Comment (antonbabenko) [inline outputs.tf]: yes, looks good to me

**Comment by antonbabenko** [comment]
Thank you @hakamadare

**Pull request merged**